src/mjolnir/ferry_connections.cc

```diff
@@ -117,11 +117,10 @@ uint32_t ShortestPath(const uint32_t start_node_idx,
         continue;
       }
 
-      // Skip destination only and uses other than road / other (service?)
+      // Skip uses other than road / other (service?)
       const OSMWay w = *ways[edge.wayindex_];
-      if (w.destination_only() ||
-          (w.use() != baldr::Use::kOther &&
-           static_cast<int>(w.use()) > static_cast<int>(baldr::Use::kTurnChannel))) {
+      if (w.use() != baldr::Use::kOther &&
```

Comment:
im wondering about this. @dnesbitt61 perhaps the correct fix is to, if we detected that while updating edges we werent able to get back to a high class road, we need to do another pass where we allow the use of destination only edges?

Reply:
ok talked this over with @dnesbitt61. instead of allowing  use of `destination_only` what we can do here is penalize it with like a 5-10 minute penalty in the cost below. this should allow us to most of the time avoid destination only expect when it is necessary.

Reply:
note that there is a break in the method once n exceeds 400. This may be too low if there is a penalty involved (may need some adjustment of this)